cellpose/gui/guiparts.py

```diff
@@ -395,6 +400,98 @@ def resizeEvent(self, event):
+
+        # Create a QGridLayout for the window
+        layout = QGridLayout()
+        layout.setContentsMargins(0, 0, 0, 0)  # Set margins (left, top, right, bottom) to zero
+        self.setLayout(layout)
+
+        # Create a widget and add the layout to it
+        self.image_widget = pg.GraphicsLayoutWidget()
+        layout.addWidget(self.image_widget, 0, 0, 1, 1)
+
+        # Load the current image from the stack
+        self.mini_image = pg.ImageItem(parent.stack[parent.currentZ])
+
+        # Create a viexbox for the minimap
+        self.viewbox = pg.ViewBox()
+        self.viewbox.setMouseEnabled(x=False, y=False)
+        # Invert and fix the aspect ratio of the viewbox to look like the original image
+        self.viewbox.invertY(True)
+        self.viewbox.setAspectLocked(True)
+        # Update the minimap so that it responds to changes in the main window
+        self.update_minimap(parent)
+        # Add the viewbox to the image widget
+        self.image_widget.addItem(self.viewbox)
+
+        # This marks the menu button as checked
+        parent.minimapWindow.setChecked(True)
+
+    def closeEvent(self, event: QEvent):
+        """
+        Method to uncheck the button in the menu if the window is closed.
+        It overrides closeEvent and is automatically called when the window is closed.
+        It calls minimap_closed from gui.py.
+        """
+        # Notify the parent that the window is closing
+        self.parent().minimap_closed()
+        event.accept()  # Accept the event and close the window
+
+    def update_minimap(self, parent):
+        """
+        Method to update the minimap.
+        It takes the current image from the parent and updates the image in the minimap.
+        """
+        if parent.img.image is not None:
+            # Obtain image data from the parent
+            image = parent.img.image
+            # Obtain saturation levels
+            levels = parent.img.levels
+            # Obtain current channel
+            current_channel = parent.color
+
+            # Set image
+            self.mini_image = pg.ImageItem(image)
+            # Display of RGB or greyscale image
+            if current_channel == 0 or current_channel == 4:
+                self.mini_image.setLookupTable(None)
+            # Display of image using a pre-defined colormap corresponding to a specific color channel
+            # (red, green, blue)
+            elif current_channel < 4:
+                self.mini_image.setLookupTable(parent.cmap[current_channel])
+            # Display of spectral mage
+            elif current_channel == 5:
+                self.mini_image.setLookupTable(parent.cmap[0])
+
+            self.mini_image.setLevels(levels)
+            self.viewbox.addItem(self.mini_image)
+
+            # Set a fixed size for the minimap
+            aspect_ratio = self.mini_image.width() / self.mini_image.height()
+            self.setFixedSize(int(400 * aspect_ratio), 400)

```

Comment:
Auch wenn es momentan nicht die Priorität in dem Ticket war: vielleicht die Größe statt hardcoded dynamisch gestalten. Dann lassen sich spätere update (z.b. adjustable sizes von der Minimap und auch dem content) leichter implementieren

Reply:
Hallo,
wir finden es auch sinnvoll, wenn sich die Größe dynamisch anpasst. Das hat jetzt Jakob in dem Brach von Ticket 11 implementiert, also diese Änderung kommt noch auf jeden Fall aber nicht auf diesem Ticket :)